pkg/oauth2/login.go

```diff
@@ -63,11 +63,22 @@ func loginCmd(vc *cmdutils.VerbCmd) {
 			"The OAuth 2.0 client identifier for pulsarctl")
 		set.StringVar(&c.Scope, "scope", c.Scope,
 			"The OAuth 2.0 scope(s) to request")
+		set.StringVar(
+			&c.AuthParams,
+			"auth-params",
+			c.AuthParams,
+			"Authentication parameters are used to configure the OAuth 2.0 provider.\n"+
+				" OAuth2 example: \"{\"audience\":\"test\",\"issuerUrl\":\"https://sample\","+
+				"\"privateKey\":\"/mnt/secrets/auth.json\",\"scope\":\"api://default/\"}\"\n")
 	})
 	vc.EnableOutputFlagSet()
 }
 
 func doLogin(vc *cmdutils.VerbCmd, config *cmdutils.ClusterConfig, noRefresh bool) error {
+	config, err := applyClientCredentialsToConfig(config)
```

Comment:
I don't suggest you change this file, their looks confused between `--auth-params` and `oauth2`.

Reply:
if not touching the oauth2 plugin files, we cannot use the `--auth-params` JSON like way to passing the configs. 
With `pulsar-admin`, we do not need to do `oauth2 active` and `oauth2 login` before running any commands to OAuth2 enabled cluster, and in SN Cloud, we use OAuth2 and some of the defined client secrets are configured as `pulsar-admin` way, so it is better to have the `pulsarctl` be able to process `pulsar-admin` like params as well.

Reply:
Ok, I see.